client/db/src/lib.rs

```diff
@@ -174,7 +174,7 @@ impl<B: BlockT> StateBackend<Blake2Hasher> for RefTrackingState<B> {
 		self.state.storage_root(delta)
 	}
 
-	fn child_storage_root<I>(&self, storage_key: &[u8], delta: I) -> (Vec<u8>, bool, Self::Transaction)
+	fn child_storage_root<I>(&self, storage_key: &[u8], delta: I) -> (H256, bool, Self::Transaction)
```

Comment:
Just curious, I do not really understand this change (all the switch of child trie api from 'Vec<u8>' to H256), it seems like it goes in the opposite direction as the pr (iirc child api was not using H::Out or H256 in prevision of using different hash type for different child tries (but there is multiple others things to change to achieve that so it could seem a bit pointless)).

Reply:
That comes in a different pr, where I completely drop the `BlockT<Hash = H256>` stuff in the codebase. However it is not that important here, as this is the `state_machine::Backend` that is a node only internface.

Reply:
> drop the BlockT<Hash = H256>
That would be amazing (I remember starting something ,but dropping the refactoring in the middle (the environmental part being tricky)) :+1: 
I aggree it is not very important.